openedx/features/content_type_gating/partitions.py

```diff
@@ -73,7 +78,91 @@ def create_content_gating_partition(course):
 
 
 class ContentTypeGatingPartition(UserPartition):
-    pass
+
+    def access_denied_fragment(self, block, user, user_group, allowed_groups):
```

Comment:
there are a number of things that I think need to change with this function.  
like moving out the html/css and changing how we check mobile and imo moving out the ecommerce stuff to a helper function, but we already have tickets for some of these things, so if we want to clean it up later that's fine  

https://openedx.atlassian.net/browse/REV-581
https://openedx.atlassian.net/browse/REV-569

Reply:
Fixed